src/panelApi.js

```diff
@@ -29,6 +29,24 @@ export async function makePanelSession(authSession, {id, uri}) {
       }
     });
 
+    socket.on('connect_error', async (err) => {
+      if (err.message === 'Authorization has failed.') {
```

Comment:
It would be better to send a constant value from the server using the `error.data` [property](https://socket.io/docs/v4/middlewares/#handling-middleware-error). Using human-readable messages for differentiating errors is quite brittle.

Reply:
Agreed, but, unfortunately, existing backends only send this error message without `data`. So, the only way to identify that this is an auth error is to check this human-readable value. Once that is supported on the server side, we can use it.

Reply:
Like I mentioned in my other comment, I noticed this PR updates socket-io to v4 which would be incompatible with existing instances of the socket-io server. If you're planning to update the server version of socket-io already, it seemed like a good oportunity to add this error metadata.